lua/focus/init.lua

```diff
@@ -1,132 +1,224 @@
+---
+---@usage `require('focus').setup({})` (replace `{}` with your `config` table)
+Focus.setup = function(config)
+    -- Export module
+    _G.Focus = Focus
+
+    -- Setup config
+    config = H.setup_config(config)
+
+    -- Apply config
+    H.apply_config(config)
+
+    if config ~= nil and config.enable then
+        autocmd.setup(config)
+
+        if config.commands then
+            commands.setup()
+        end
+
+        if config.ui.winhighlight then
+            if vim.fn.has('nvim-0.9') then
+                vim.api.nvim_set_hl(0, 'FocusedWindow', { link = 'VertSplit' })
+                vim.api.nvim_set_hl(0, 'UnfocusedWindow', { link = 'Normal' })
+            else
+                vim.highlight.link('FocusedWindow', 'VertSplit', true)
+                vim.highlight.link('UnfocusedWindow', 'Normal', true)
+            end
+
+            vim.wo.winhighlight =
+                'Normal:FocusedWindow,NormalNC:UnfocusedWindow'
+        end
+
+        if config.autoresize then
+            Focus.resize()
+        end
+    end
+end
+
+Focus.resize = function()
+    local config = H.get_config()
+
+    resizer.split_resizer(config)
 end
 
---Used to run commands in respect to filetype/buffertype exclusions
-M.run_cmd = function(command)
-	autocmd.run_cmd(M, command)
+-- Exported internal functions for use in commands
+function Focus.split_nicely(args)
+    local config = H.get_config()
+    if args == nil then
+        args = ''
+    end
+    split.split_nicely(args, config.bufnew)
 end
 
-M.resize = function()
-	resizer.split_resizer(M)
+function Focus.split_command(direction, args)
+    local config = H.get_config()
+    args = args or ''
+    split.split_command(direction, args, config.split.tmux, config.split.bufnew)
 end
 
--- Exported internal functions for use in commands
-function M.split_nicely(args)
-	split.split_nicely(args, M.bufnew)
+function Focus.split_cycle(reverse)
+    local config = H.get_config()
+    split.split_cycle(reverse, config.bufnew)
 end
 
-function M.split_command(direction, args)
-	args = args or ''
-	split.split_command(direction, args, M.tmux, M.bufnew)
+function Focus.focus_enable()
+    functions.focus_enable()
 end
 
-function M.split_cycle(reverse)
-	split.split_cycle(reverse, M.bufnew)
+function Focus.focus_disable()
+    functions.focus_disable()
 end
 
-function M.focus_enable()
-	functions.focus_enable()
+function Focus.focus_toggle()
+    functions.focus_toggle()
 end
 
-function M.focus_disable()
-	functions.focus_disable()
+function Focus.focus_maximise()
+    functions.focus_maximise()
 end
 
-function M.focus_toggle()
-	functions.focus_toggle()
+function Focus.focus_equalise()
+    functions.focus_equalise()
 end
 
-function M.focus_maximise()
-	functions.focus_maximise()
+function Focus.focus_max_or_equal()
+    functions.focus_max_or_equal()
 end
 
-function M.focus_equalise()
-	functions.focus_equalise()
+function Focus.focus_disable_window()
```

Comment:
It may be a good idea to reset disabled windows when `focus_enable` and `focus_disable` are called. Individually disabling a window and then calling `focus_disable` can result in autoresizing breaking entirely until `focus_enable_window` is called for each window that was individually disabled.